platform/ios/src/MGLMapView.mm

```diff
@@ -4150,6 +4173,111 @@ - (void)applyIconIdentifier:(NSString *)iconIdentifier toAnnotationsWithImageReu
     }
 }
 
+#pragma mark - Shape Annotation
+
+- (void)selectShapeAnnotation:(MGLShape *)shapeAnnotation
+{
+    if (!shapeAnnotation) return;
+    
+    if (shapeAnnotation == _selectedShapeAnnotation) return;
+    
+    [self deselectShapeAnnotation:_selectedShapeAnnotation];
+    
+    _selectedShapeAnnotation = shapeAnnotation;
+    
+    if ([self.delegate respondsToSelector:@selector(mapView:didSelectShapeAnnotation:)])
+    {
+        [self.delegate mapView:self didSelectShapeAnnotation:shapeAnnotation];
+    }
+}
+
+- (void)deselectShapeAnnotation:(MGLShape *)shapeAnnotation
+{
+    if (!shapeAnnotation) return;
+    
+    if (_selectedShapeAnnotation == shapeAnnotation)
+    {
+        if ([self.delegate respondsToSelector:@selector(mapView:didDeselectShapeAnnotation:)])
+        {
+            [self.delegate mapView:self didDeselectShapeAnnotation:shapeAnnotation];
+        }
+        _selectedShapeAnnotation = nil;
+    }
+   
+}
+
+- (MGLShape*)shapeAnnotationForGestureRecognizer:(UITapGestureRecognizer*)singleTap
+{
+    CGPoint tapPoint = [singleTap locationInView:self];
+//    CGRect queryRect = CGRectInset({ tapPoint, CGSizeZero },
```

Comment:
These commented lines were pulled into methods. Don't forget to delete them here.